### PR TITLE
Task/part6 attribute value render

### PR DESCRIPTION
**Pull request by kzangeli**
Part 6 of #1259 - rendering of attribute value

### CODE REVIEW

File: src/lib/ngsi/ContextAttribute.h [hunk 1]
```diff
@@ -85,6 +85,7 @@ typedef struct ContextAttribute
   std::string  renderAsJsonObject(ConnectionInfo* ciP, RequestType request, const std::string& indent, bool comma, bool omitValue = false);
   std::string  renderAsNameString(ConnectionInfo* ciP, RequestType request, const std::string& indent, bool comma = false);
   std::string  toJson(bool isLastElement, bool types, const std::string& renderMode);
+  std::string  toJsonAsValue(ConnectionInfo* ciP);
```
Comment (fgalan) [inline src/lib/ngsi/ContextAttribute.h]: What's the different between toJson() and toJsonAsValue() ?

File: test/functionalTest/cases/0000_content_related_headers/in_out_formats.test [hunk 2]
```diff
@@ -86,12 +86,12 @@ echo '================================================================='
 echo
 
 echo "1.4: Unrecognized Content-Type, Accepts text/plain: 415, response in XML"
-orionCurl --url "$url" --payload "${payload}" --in "text/plain" --out "text/plain"
+orionCurl --url "$url" --payload "${payload}" --in "text/plain" --out "text/plain" --payloadCheck xml
```
Comment (fgalan) [inline test/functionalTest/cases/0000_content_related_headers/in_out_formats.test]: Why this change was needed in such old .test?

File: test/functionalTest/cases/0997_GET_v2_entity_single_attribute_value/GET_v2_entity_single_attribute_value.test [hunk 3]
```diff
@@ -180,65 +180,41 @@ Date: REGEX(.*)
...
 Date: REGEX(.*)
 
-{
-    "value": {
-        "metadata": {},
-        "type": null,
-        "value": "Mercedes Benz"
-    }
-}
+Mercedes Benz
```
Comment (fgalan) [inline test/functionalTest/cases/0997_GET_v2_entity_single_attribute_value/GET_v2_entity_single_attribute_value.test]: Should this be

```
Mercedes Benz
```

or

```
"Merces Benz"
```

?

In order to kep "symmetry" with PUT operation, it should have "". In order to keep simpler, it should be without "".

I don't have an answer for this :) Please, cast the question in the NGSIv2 skype chat spec.

File: test/functionalTest/cases/1179_options_test/GET_v2_ent_attr_val_options_text.test [hunk 4]
```diff
@@ -42,49 +42,56 @@ echo "====================="
 payload='{
   "id": "E1",
   "type": "T1",
-  "a_string":"some value",
```
Comment (fgalan) [inline test/functionalTest/cases/1179_options_test/GET_v2_ent_attr_val_options_text.test]: Only indentation changes in this part, right?

File: src/lib/ngsi/ContextAttribute.h (hunk 1 above)
Comment (kzangeli) [inline src/lib/ngsi/ContextAttribute.h]: Normally ContextAttribute is rendered as 

    "name": {
       "type": "x",
       "value": "123",
      ...

This new method I added for the case  "GET /v2/entities/EEE/attr/attrrName/value", where only the value is rendered (and as standalone).
I could have added it to **toJson()**, with a new parameter, but it would make that method innecessarily complex.

File: test/functionalTest/cases/0000_content_related_headers/in_out_formats.test (hunk 2 above)
Comment (kzangeli) [inline test/functionalTest/cases/0000_content_related_headers/in_out_formats.test]: Fixes in orionCurl.
If out format is json or xml, orionCurl does the payloadCheck (beautifier).
In this case the out-format is text/plain, but orionCurl STILL did the payloadCheck.
The new outputs for text/plain, that are just a string or a Number fail, of course.
I had to modify the *payload check mechanism* in orionCurl.
Here, the output is actually XML, and I want to force the payload check mechanism.

File: test/functionalTest/cases/0997_GET_v2_entity_single_attribute_value/GET_v2_entity_single_attribute_value.test (hunk 3 above)
Comment (kzangeli) [inline test/functionalTest/cases/0997_GET_v2_entity_single_attribute_value/GET_v2_entity_single_attribute_value.test]: Yeah, I was also unsure about this.

Would it not be nice to have:
  Content-Type: text/number
or something similar?

"Mercedes Benz" is not very good (apart from being a bit ugly), as the users will have to strip the citation marks ...

File: test/functionalTest/cases/1179_options_test/GET_v2_ent_attr_val_options_text.test (hunk 4 above)
Comment (kzangeli) [inline test/functionalTest/cases/1179_options_test/GET_v2_ent_attr_val_options_text.test]: Now that I had to modifiy this file, I fixed the indentations also. It was ugly ...

The real change is that **?options=text** is removed and that the response is in JSON for compound values

**Comment by kzangeli** [comment]
LGTM

File: src/lib/ngsi/ContextAttribute.h (hunk 1 above)
Comment (fgalan) [inline src/lib/ngsi/ContextAttribute.h]: NTC

File: test/functionalTest/cases/0000_content_related_headers/in_out_formats.test (hunk 2 above)
Comment (fgalan) [inline test/functionalTest/cases/0000_content_related_headers/in_out_formats.test]: NTC

File: test/functionalTest/cases/0997_GET_v2_entity_single_attribute_value/GET_v2_entity_single_attribute_value.test (hunk 3 above)
Comment (fgalan) [inline test/functionalTest/cases/0997_GET_v2_entity_single_attribute_value/GET_v2_entity_single_attribute_value.test]: Already discussed. NTC in this PR

File: test/functionalTest/cases/1179_options_test/GET_v2_ent_attr_val_options_text.test (hunk 4 above)
Comment (fgalan) [inline test/functionalTest/cases/1179_options_test/GET_v2_ent_attr_val_options_text.test]: NTC

**Comment by fgalan** [comment]
LGTM
